runtime/shared_heap.c

```diff
@@ -453,16 +450,36 @@ static void* pool_allocate(struct caml_heap_state* local, sizeclass sz) {
   if (!r) return 0;
 
   p = r->next_obj;
-  next = (value*)p[1];
+  /* assert that p is inside the pool */
+  CAMLassert(p >= (value*)r + POOL_HEADER_WSIZE);
```

Comment:
Could be POOL_FIRST_BLOCK

Reply:
Resolved in https://github.com/ocaml/ocaml/pull/13616/commits/29be98829b078a50103cfb30ce1b71830bc7cb92